### PR TITLE
Fix NPE in regex parser with single quantifiers

### LINKED ISSUE #40541: [Bug]: NPE when formatting Ballerina code that includes regular expressions
### Description

Occurred this NPE errors[2] while formatting the below ballerina code[1] using `bal format` command

[1] sample Ballerina code
```ballerina
import ballerina/constraint;

@constraint:String {pattern: re `[A-Z]{1}`}
public type FlightClass string;

@constraint:String {pattern: re `[a-zA-Z0-9]{3}`}
public type Code string;

public function main() {
    Code code = "A380";
    FlightClass flightclass = "A";
}

```
[2] Error stack
```
May 30, 2023 8:57:24 PM org.ballerinalang.formatter.core.FormattingTreeModifier formatNode
SEVERE: Error while formatting [kind: RE_SEQUENCE] [line: 3] [column:33]: java.lang.NullPointerException
May 30, 2023 8:57:24 PM org.ballerinalang.formatter.core.FormattingTreeModifier formatNode
SEVERE: Error while formatting [kind: RE_SEQUENCE] [line: 6] [column:33]: java.lang.NullPointerException
modified files:
/Users/hansaninissanka/Documents/Ballerina Projects/pattern_connector/main.bal

format successful.
``` 

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.5.0
2201.6.0-rc1

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #40541 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    Code2 code2 = "A380";
+    FlightClass2 flightclass2 = "A";
+
+    Code3 code3 = "A380";
+    FlightClass3 flightclass3 = "A";
+
+    Code4 code4 = "A380";
+    FlightClass4 flightclass4 = "A";
+}
```
Comment (pcnfernando) [inline misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal:39]: Can you add few tests without using contraint module

Comment (SasinduDilshara) [reply misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal:37]: Added

**Review by pcnfernando — DISMISSED** [review]
LGTM

File: misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import ballerina/constraint;
```
Comment (MaryamZi) [inline misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal:1]: We don't use standard library modules in lang tests.

1. introduces a cyclic dependency between lang and the stdlib module
2. can result in build failures
 i. due to bug fixes that cause the stdlib module builds to fail - need to wait for them to be fixed
 ii. local build failures due to incompatible BIR caches

Comment (SasinduDilshara) [reply misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal]: please check now.

File: misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+public function test1() {
+    Foo _={name: "abc"};
+    Foo2 _={name: "abc"};
+    Foo3 _={name: "abc"};
+}
+
+public function test2() {
+    string:RegExp _=re `[A-Z]{1}`;
```
Comment (pcnfernando) [inline misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal:35]: Check the formatting

File: misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    string:RegExp _=re `[A-Z]{1,2}`;
+    string:RegExp _=re `[A-Z]{1,}`;
+}
```
Comment (pcnfernando) [inline misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal:38]:
```suggestion
type AnnotRec record {|
    string:RegExp value;
|};

annotation AnnotRec annot on type;

@annot {
    value: re `abc{1}`
}
type Foo record {
    string name;
};

@annot {
    value: re `abc{1,}`
}
type Foo2 record {
    string name;
};

@annot {
    value: re `abc{1,2}`
}
type Foo3 record {
    string name;
};

public function test1() {
    Foo _ = {name: "abc"};
    Foo2 _ = {name: "abc"};
    Foo3 _ = {name: "abc"};
}

public function test2() {
    string:RegExp _ = re `[A-Z]{1}`;
    string:RegExp _ = re `[A-Z]{1,2}`;
    string:RegExp _ = re `[A-Z]{1,}`;
}
```

File: misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal (hunk 3 above)
Comment (SasinduDilshara) [reply misc/formatter/modules/formatter-core/src/test/resources/expressions/raw-template/source/regex_expressions.bal]: The wrong formatting is done intentionally to check the correct behaviour in formate

**Review by pcnfernando — APPROVED** [review]

**Comment by pcnfernando** [comment]
@SasinduDilshara please port this fix to the relevant patch branches

**Pull request merged**